src/app/hooks/useUserRights.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
+import { client } from '../lib/http-client';
+import { configuration } from '../lib/configuration';
+import {
+  RightToOrganization,
+  UserRights,
+  AUTH_SERVICE,
+  RIGHT,
+  getOrgsForRight,
+} from '../lib/auth';
+
+export const getUserRights = async (): Promise<UserRights> => {
+  const response = await client.get(configuration.kayttoikeusUrl);
+  if (!response.data) {
+    throw Error('Unable to get user data');
+  }
+  const organizations: RightToOrganization[] = response.data.organisaatiot
+    .map(
+      (o: {
+        organisaatioOid: string;
+        kayttooikeudet: [{ palvelu: string; oikeus: RIGHT }];
+      }) => {
+        const rights: RIGHT[] = o.kayttooikeudet
+          .filter((o) => o.palvelu === AUTH_SERVICE)
+          .map((o) => o.oikeus);
+        return rights.length > 0
+          ? { organizationOid: o.organisaatioOid, rights }
+          : null;
+      },
+    )
+    .filter((o: RightToOrganization | null) => o !== null);
+  const userRights: UserRights = {
+    admin: response.data.isAdmin,
+    readOrganizations: getOrgsForRight(organizations, 'READ'),
+    writeOrganizations: getOrgsForRight(organizations, 'READ_UPDATE'),
+    crudOrganizations: getOrgsForRight(organizations, 'CRUD'),
+  };
+  if (userRights.readOrganizations.length < 1 && !userRights.admin) {
```

Comment:
Eikö `=== 0` ole tässä sama asia kuin `< 1`?
```suggestion
  if (userRights.readOrganizations.length === 0 && !userRights.admin) {
```

Reply:
Se on js:n kehittämää vainoharhaisuutta, tarkistan pienempi kuin koska -1 menisi === 0 tarkistuksesta läpi.